### PR TITLE
add overview and quickstart to README (master)

### PR DESCRIPTION
**Pull request by trel**

### CODE REVIEW

**Comment by trel** [comment]
@korydraughn is the quickstart too much redundant information?    i feel it's necessary, but... the information is now in two places.  i feel strongly both ways.

File: README.md [hunk 1]
```diff
@@ -1,5 +1,40 @@
 # iRODS C++ REST Mid-Tier API
 
+This REST API is designed to be run alongside an iRODS Server to provide an HTTP REST interface to the majority of administrative functions for an iRODS Zone.
```
Comment (korydraughn) [inline README.md:3]: **administrative** makes it sound like the REST API only provides admin level operations.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,40 @@
...
+
+It will run under the same linux service account and access `/etc/irods/server_config.json` and the active authenticated `rodsadmin` iRODS account.
+
+This REST API requires an iRODS Server v4.2.5 or greater.
```
Comment (korydraughn) [inline README.md:7]: 4.2.5 may be too early.

This project uses (without scanning every line of code):
- query iterator
- dstream
- filesystem
- resource administration
- at_scope_exit

There might be other libs it uses as well. I'll check and report back here.

File: README.md (hunk 1 above)
Comment (trel) [reply README.md:3]: oh right, that's ZMT...   rewording...

File: README.md (hunk 2 above)
Comment (trel) [reply README.md:7]: Ah, right - the limiting factor is the 'latest' of the used libraries, not the oldest.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,40 @@
 # iRODS C++ REST Mid-Tier API
 
+This REST API is designed to be run alongside an iRODS Server to provide an HTTP REST interface to the majority of administrative functions for an iRODS Zone.
+
+It will run under the same linux service account and access `/etc/irods/server_config.json` and the active authenticated `rodsadmin` iRODS account.
```
Comment (korydraughn) [inline README.md:5]: **will** makes the statement sound optional.

The REST API must run under the service account using the authenticated `rodsadmin` account. I feel we need a name that clearly identifies the **first iRODS account** as the **super account**.

Comment (trel) [reply README.md]: removed the 'will'

Comment (trel) [reply README.md]: it doesn't have to be the first/super irods account... just the active rodsadmin one being used.

**Review by korydraughn — COMMENTED** [review]
Yes, a lot of this is redundant. However, it doesn't hurt to have a quickstart section that allows people to stand things up as quickly as possible.

Then again, this could be captured in docker-compose. Then no one needs to know the steps. We just expose some knobs that allow people to tweak it and we're done.

With that said, maybe we replace the quickstart contents with steps for standing up a containerized version?

**Comment by trel** [comment]
Hmmm, possibly...   not sure how a containerized version can interoperate with an existing syslog or `/etc/irods/server_config.json`...    just volume mounts?

**Comment by korydraughn** [comment]
I can try making a compose project just to see if it's possible. The current quickstart section can move forward as-is if you're happy with it. We can always update the README as things change.

**Comment by trel** [comment]
gotcha - makes sense.

File: README.md (hunk 2 above)
Comment (trel) [reply README.md]: Going with 4.2.0 for now...

This is not about the implementations within each of irods_client_rest_cpp or the server itself... it's about iRODS protocol compatibility and, I think... the `resc_name` to `resc_id` conversion in 4.2.0.

**Pull request merged**